libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -23,6 +23,10 @@ logScope:
 
 when defined(libp2p_network_protocols_metrics):
   declareCounter libp2p_protocols_bytes, "total sent or received bytes", ["protocol", "direction"]
+  declareHistogram libp2p_protocols_qlen, "message queue length", ["protocol"],
```

Comment:
Does it really make sense to have theses metrics be per protocol?
Since the queue is global, they should be about the same for every protocol, no?

Reply:
The queue should be per stream tho, which can be mapped to a protocol.

Reply:
Having it be per-protocol means 12 metrics instead of one, we have to be sure it's worth it

Reply:
You're right, there are limits to how many metrics we can collect in practice, even for tracing/debugging purposes, for example prometheus gets quickly overwhelmed if too many metrics are being pushed to it.